app/components/filters/date-range-filter.ts

```diff
@@ -1,63 +1,189 @@
 import { action } from '@ember/object';
-import FilterComponent from './filter';
+import RouterService from '@ember/routing/router-service';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
+import {
+  endOfMonth,
+  endOfWeek as endOfWeekDateFns,
+  endOfYear,
+  formatISO,
+  startOfMonth,
+  startOfWeek as startOfWeekDateFns,
+  startOfYear,
+  sub,
+} from 'date-fns';
 
-export default class SearcherDateRangeFilterComponent extends FilterComponent {
-  @tracked start: string | undefined = this.args.valueA
-    ? this.args.valueA
-    : undefined;
-  @tracked end: string | undefined = this.args.valueB
-    ? this.args.valueB
-    : undefined;
+type ISODateString = string;
 
-  changeValue(start?: string, end?: string) {
+interface Signature {
+  Args: {
+    id: string;
+    startQueryParam: string;
+    endQueryParam: string;
+    start?: ISODateString;
+    end?: ISODateString;
+  };
+}
+
+enum Presets {
+  ThisWeek = 'Deze week',
+  LastWeek = 'Voorbije week',
+  ThisMonth = 'Deze maand',
+  LastMonth = 'Vorige maand',
+  ThisYear = 'Dit jaar',
+  LastYear = 'Vorig jaar',
+  CustomRange = 'Aangepaste periode',
```

Comment:
Still WIP and the specifics are still being worked out. I didn't want to drop everything I had though, since some things can probably be used still.